pyproject.toml

```
@@ -756,4 +756,20 @@ required-environments = [
 constraint-dependencies = [
     "asdf>=2.12.0",
     "asdf-coordinates-schemas>=0.2.0",
+
+    # reproduce IPython minimal requirements on its dependencies
+    # as of version 9.7.0, the earliest one to consistently declare
+    # and test its lower bounds.
+    # this entire chunk may be removed when we drop support for IPython<9.7
+    'colorama>=0.4.4; sys_platform == "win32"',
+    "decorator>=4.3.2",
+    "ipython-pygments-lexers>=1.0.0",
+    "jedi>=0.18.1",
+    "matplotlib-inline>=0.1.5",
+    'pexpect>4.3; sys_platform != "win32" and sys_platform != "emscripten"',
+    "prompt_toolkit>=3.0.41,<3.1.0",
```

Why should we want to set a constraint from the above?
```suggestion
    "prompt_toolkit>=3.0.41",
```